http-server/src/main/java/org/triplea/server/user/account/login/authorizer/legacy/Md5Crypt.java

```diff
@@ -0,0 +1,98 @@
+package org.triplea.server.user.account.login.authorizer.legacy;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.apache.commons.codec.digest.Md5Crypt.md5Crypt;
+
+import java.nio.charset.StandardCharsets;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A collection of methods for using the FreeBSD MD5-crypt hash algorithm.
+ *
+ * @see <a
+ *     href="https://www.usenix.org/legacyurl/md5-crypt">https://www.usenix.org/legacyurl/md5-crypt</a>
+ * @see <a href="https://www.systutorials.com/docs/linux/man/n-md5crypt/">
+ *     https://www.systutorials.com/docs/linux/man/n-md5crypt/ </a>
+ */
+final class Md5Crypt {
+  private static final String MAGIC = "$1$";
+  private static final Pattern HASHED_VALUE_PATTERN =
+      Pattern.compile(
+          "^" + MAGIC.replace("$", "\\$") + "([\\.\\/a-zA-Z0-9]{1,8})\\$([\\.\\/a-zA-Z0-9]{22})$");
```

Comment:
Might be worth considering using `Pattern.quote(MAGIC)`

Reply:
I should have called out better that Md5Crypte is legacy/copied code. Ideally we would delete it soon. Given it's not tested, I'm reluctant to modify it.

Reply:
(Though, there will be an integration test landing, so that might be some confidence to modify it, but again because it's copied, and also temporary/legacy, there is some value in leaving it as-is too).

Reply:
Yeah I figured, I mean the only reason that string is escaped is because `$` has a special meaning in regex